pkg/crd/markers/package.go

```diff
@@ -22,7 +22,7 @@ import (
 
 func init() {
 	AllDefinitions = append(AllDefinitions,
-		must(markers.MakeDefinition("groupName", markers.DescribesPackage, "")).
```

Comment:
@sttts Does it sound reasonable for you to just adjust the marker here to accept an "empty" groupName marker ("// +groupName=") without any additional handling for the empty string

Reply:
Context: Without this PR controller-gen fails like this now (with API types that use types from k8s.io/api@v0.31.0-alpha.3)

> /home/prow/go/pkg/mod/k8s.io/api@v0.31.0-alpha.3/core/v1/doc.go:21:1: missing argument "" (at <input>)
> /home/prow/go/pkg/mod/k8s.io/api@v0.31.0-alpha.3/core/v1/doc.go:21:1: missing argument "" (at <input>)
> https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/10803/pull-cluster-api-verify-main/1811372226943913984

Reply:
> without any additional handling for the empty string?

This is about embedding corev1 types in some CRD, right? So the groupName of the corev1 package does not really play a role for the CRD, does it? So yes, I think just parsing it without falling over, and then ignoring it outside of actual CRD API package sounds right.

Reply:
> This is about embedding corev1 types in some CRD, right?

Yes exactly

> So the groupName of the corev1 package does not really play a role for the CRD, does it?

This is my understanding, yes

Reply:
I'm not sure if anyone is doing it, but theoretically folks could use controller-gen to generate CRD for types of the k/k core API. In that case they would end up with an empty group in the CRD with `groupName=` (see the test in this PR: pkg/crd/testdata/testdata._cronjobs.yaml)

Not sure which group would be correct in this (probably hypothetical) case. But folks could just specify whatever group they want via groupName, so I assume it's fine

Reply:
Empty string would be correct for the group name, `"core"` for the group in the CRD name. At least that's what kcp does: https://github.com/kcp-dev/kcp/blob/main/test/e2e/fixtures/kube/core.k8s.io_pods.yaml#L9

Reply:
Okay, thx! 

I think then let's do the same in controller-gen for now. We can still modify this behavior later (or offer some additional markers) if people want something else for CRD generation for the core API group

Reply:
@twz123 WDYT about handling the empty group here: https://github.com/kubernetes-sigs/controller-tools/blob/b370e9b1e3394f945fe796accc424569a1a62892/pkg/crd/spec.go#L73 to use "core" for the CRD name suffix if .Group is an empty string?

Reply:
@sttts I was trying to deploy a CRD with name & group like this, but I'm getting these errors with a regular kube-apiserver (kindest/node:v1.30.0)
```
The CustomResourceDefinition "cronjobs.core" is invalid:
* metadata.name: Invalid value: "cronjobs.core": must be spec.names.plural+"."+spec.group
* spec.group: Required value
```

CRD:
```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: (devel)
  name: cronjobs.core
spec:
  group: ""
  names:
    kind: CronJob
    listKind: CronJobList
    plural: cronjobs
    singular: cronjob
  scope: Namespaced
...
```

Reply:
In Kube does is not allowed today. You cannot mix native and CRD api groups.